tmk_core/common/action_util.c

```diff
@@ -21,6 +21,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "action_layer.h"
 #include "timer.h"
 #include "keycode_config.h"
+#include <quantum_keycodes.h>
```

Comment:
I think this is causing issues in Travis CI for ARM boards. 

Maybe we need:
```suggestion
#ifdef ___AVR___
#include <quantum_keycodes.h>
#endif
```

Reply:
This won't help, I'm including the file to access the QK_* mod macro definitions and so it won't compile on arm either. The problem is that the RESET keycode in quantum_keycodes.h conflicts with the one in stm32f3xx.h file which is itself included due to

```
#ifdef PROTOCOL_CHIBIOS
#include "hal.h"
#endif
```
in usb_descriptor.h. This, in turn, seems to only be required for
```
#define MAX_ENDPOINTS USB_MAX_ENDPOINTS
```

At line 215. USB_MAX_ENDPOINTS is only used in the next check
```
#if (NEXT_EPNUM - 1) > MAX_ENDPOINTS
# error There are not enough available endpoints to support all functions. Remove some in the rules.mk file. (MOUSEKEY, EXTRAKEY, CONSOLE, NKRO, MIDI, SERIAL, STENO)
#endif
```

So for solving this problem, we might:

1. Rename one of the RESET enum so that they don't conflict anymore,
2. Move the new code out of action_util.c if there is a better place,
2. Put the check for available endpoints and the related #include in the usb_descriptor.c file so that the smt32f3xx.h RESET symbol is not exported to the other object file.

I think that the second alternative is the cleaner one. No need to export symbols that are not needed so I will update the PR with this change. I've tested by building for planck/rev4 and planck/rev6. If you prefer any of the other option just tell me but the first one might be touching a lot of files anyway.